### PR TITLE
cdef.d: harmonize with backend

### PR DESCRIPTION
**Pull request by WalterBright**
In converting DMC++ source to D, I had to change somethings in the headers. This harmonizes them.

### CODE REVIEW

**Comment by JinShil** [comment]
I tried to review this, but I have no idea what's going on here.  There's a lot of stuff that was commented out, and now isn't.  Was this file used in some automation script that uncommented certain lines under certain conditions, of something like that?

File: src/dmd/backend/cdef.d [hunk 1]
```diff
@@ -22,11 +22,36 @@ extern (C++):
...
+    }
+
+    enum IN_GCC     = xversion!`IN_GCC`;
+
+    enum TARGET_LINUX   = xversion!`linux`;
+    enum TARGET_OSX     = xversion!`OSX`;
+    enum TARGET_FREEBSD = xversion!`FreeBSD`;
+    enum TARGET_OPENBSD = xversion!`OpenBSD`;
+    enum TARGET_SOLARIS = xversion!`Solaris`;
+    enum TARGET_WINDOS  = xversion!`Windows`;
```
Comment (dkgroot) [inline src/dmd/backend/cdef.d]: As DragonFlyBSD Support was merged yesterday, shouldn't it also be specified here ?
```
enum TARGET_DRAGONFLYBSD = xversion!`DragonFly`;
```

**Review by dkgroot — CHANGES_REQUESTED** [review]

File: src/dmd/backend/cdef.d [hunk 2]
```diff
@@ -22,11 +22,36 @@ extern (C++):
...
+    version = XVERSION;
+
+version (XVERSION)
+{
+    extern (D) template xversion(string s)
+    {
+        enum xversion = mixin(`{ version (` ~ s ~ `) return true; else return false; }`)();
+    }
+
+    enum IN_GCC     = xversion!`IN_GCC`;
```
Comment (ibuclaw) [inline src/dmd/backend/cdef.d]: You shouldn't need this.

**Review by ibuclaw — DISMISSED** [review]

**Comment by WalterBright** [comment]
What happened was the commented out stuff was for the DMC++ front end. Now that I've converted that to D, it can be uncommented.

File: src/dmd/backend/cdef.d (hunk 1 above)
Comment (WalterBright) [reply src/dmd/backend/cdef.d]: done - see next line

**Review by ibuclaw — APPROVED** [review]

**Comment by ibuclaw** [comment]
> I tried to review this, but I have no idea what's going on here. There's a lot of stuff that was commented out, and now isn't. Was this file used in some automation script that uncommented certain lines under certain conditions, or something like that?

From what I can tell, the dmd backend diverged between dmd, dmc, and htod during the conversion to D, and this just corrects that. None of this affects dmd itself.

**Pull request merged**